Bio/Restriction/Restriction.py

```diff
@@ -938,22 +938,20 @@ def _search(cls):
 
         Implement the search method for non palindromic enzymes.
         """
-        compsite_for, compsite_rev = cls.compsite.pattern.split('|')
-        iterator_for = cls.dna.finditer(compsite_for, cls.size)
-        iterator_rev = cls.dna.finditer(compsite_rev, cls.size)
+        iterator = cls.dna.finditer(cls.compsite, cls.size)
         cls.results = []
         modif = cls._modify
         revmodif = cls._rev_modify
         s = str(cls)
         cls.on_minus = []
 
-        for start, group in iterator_for:
+        for start, group in iterator:
             if group(s):
                 cls.results += [r for r in modif(start)]
+            else:
+                cls.on_minus += [r for r in revmodif(start)]
```

Comment:
Does this alter the order in which motifs on the reverse strand are reported? I am assuming everything is fine as the tests pass, but wanted to ask just in case.

Reply:
No, ```cls.results``` and ```cls.on_minus``` are combined afterwards and then sorted. Actually, this is just a reversion of an earlier commit of mine, where I separated the forward and reverse search, because I realized that a 'hit' on one strand prevented a 'hit' on the other strand. At that time 'lookahead search' was not in my vocabulary and indeed the test case was an overlapping recognition sequence

Reply:
Thanks for the clarification.